From 6f8c28d4ad7d4cfcee2a262ac1dc8032a3f7314b Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Wed, 14 Oct 2020 12:47:21 +0300
Subject: [PATCH] [SW-21652] goya2: remove sram scrambling init

in case of legacy boot flow, preboot gets loaded into SRAM and
scrambling init can't be done in preboot. it should be moved to BTL

for arcboot rom flow, sram scrambling init can happen in preboot as
preboot is loaded into scratchpad sram

Change-Id: I321e2eb1010e4dfb1d618fc187b2a6b4b90928a3
Signed-off-by: Sailaja <sbandarupalli@habana.ai>
---

diff --git a/goya2/asic.c b/goya2/asic.c
index fe1f864..972527c 100644
--- a/goya2/asic.c
+++ b/goya2/asic.c
@@ -7,14 +7,6 @@
 
 #include <psoc_global_conf_regs.h>
 #include <psoc_gpio0_regs.h>
-#include <dcon0_hbw_rtr_if0_rtr_h3_regs.h>
-#include <dcon0_hbw_rtr_if1_rtr_h3_regs.h>
-#include <dcon1_hbw_rtr_if0_rtr_h3_regs.h>
-#include <dcon1_hbw_rtr_if1_rtr_h3_regs.h>
-#include <dcon2_hbw_rtr_if0_rtr_h3_regs.h>
-#include <dcon2_hbw_rtr_if1_rtr_h3_regs.h>
-#include <dcon3_hbw_rtr_if0_rtr_h3_regs.h>
-#include <dcon3_hbw_rtr_if1_rtr_h3_regs.h>
 
 #include <preboot.h>
 #include <pll.h>
@@ -69,17 +61,4 @@
 	val = regmap_readl(mmPSOC_GPIO0_DATAOUT);
 	val &= ~BIT(2);
 	regmap_writel(val, mmPSOC_GPIO0_DATAOUT);
-}
-
-/* initialize sram scrambling before u-boot is downloaded */
-void sram_scrambling_init(void)
-{
-	regmap_writel(1, mmDCON0_HBW_RTR_IF0_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON0_HBW_RTR_IF1_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON1_HBW_RTR_IF0_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON1_HBW_RTR_IF1_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON2_HBW_RTR_IF0_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON2_HBW_RTR_IF1_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON3_HBW_RTR_IF0_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON3_HBW_RTR_IF1_RTR_H3_SCRAM_SRAM_EN);
 }
